### PR TITLE
misc http segment discovery fixes

### PR DESCRIPTION
**Pull request by himanshug**
Fixes following
1) getInventory() impl inadvertantly leaked `servers` hashmap, it is updated to be ConcurrentHashMap for safety.
2) suppress error logging when queryable server is not reachable for a configurable duration (default 1 minute) , it happens regularly during MM upgrades. changes are made to print info messages without exceptions in that duration.

### CODE REVIEW

File: server/src/main/java/io/druid/client/HttpServerInventoryView.java [hunk 1]
```diff
@@ -403,13 +387,11 @@ public boolean isStarted()
...
   {
-    synchronized (servers) {
-      DruidServerHolder holder = servers.get(serverKey);
-      if (holder != null) {
-        return holder.druidServer.getSegment(segment.getIdentifier()) != null;
-      } else {
-        return false;
-      }
+    DruidServerHolder holder = servers.get(serverKey);
+    if (holder != null) {
```
Comment (cheddar) [inline server/src/main/java/io/druid/client/HttpServerInventoryView.java]: This is a nit, but I find inverted logic like this to be hard to following when reading code.  I think it would be better as just

```
if (holder == null) {
  return false;
}

return holder.druidServer.getSegment(segment.getIdentifier()) != null;
```

Comment (himanshug) [inline server/src/main/java/io/druid/client/HttpServerInventoryView.java]: changed.

File: server/src/main/java/io/druid/client/HttpServerInventoryViewConfig.java [hunk 2]
```diff
@@ -24,25 +24,39 @@
...
+  @JsonProperty
+  private final long serverUnstabilityTimeout;
+
   @JsonProperty
   private final int numThreads;
 
   @JsonCreator
   public HttpServerInventoryViewConfig(
       @JsonProperty("serverTimeout") Period serverTimeout,
+      @JsonProperty("serverUnstabilityTimeout") Period serverUnstabilityTimeout,
```
Comment (b-slim) [inline server/src/main/java/io/druid/client/HttpServerInventoryViewConfig.java]: wondering would this be ok to ser/des test when the variable has 2 different types ?

Comment (himanshug) [inline server/src/main/java/io/druid/client/HttpServerInventoryViewConfig.java]: this object is only ever deserialized from config, it would never be serialized and then deserialized back so different types are ok in this context.

Comment (himanshug) [inline server/src/main/java/io/druid/client/HttpServerInventoryViewConfig.java]: that said, yeah , having a test to at least test the deserialization is good, so i will add that shortly.

Comment (himanshug) [inline server/src/main/java/io/druid/client/HttpServerInventoryViewConfig.java]: @b-slim added test
